### PR TITLE
Docs/add compatibility matrix

### PR DESCRIPTION
**Pull request by jboreiko**
This needs a few additions, but thought it might be valuable to maintain for our supported KVSs since there doesn't seem to be a list otherwise.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1189)

### CODE REVIEW

File: docs/source/services/index.rst [hunk 1]
```diff
@@ -10,4 +10,5 @@ each of the services as well as the guarentees they uphold.
    :maxdepth: 2
    :titlesonly:
 
+   key_value_stores/index
```
Comment (gsheasby) [inline docs/source/services/index.rst]: why the confusion between "key value stores" and "key value services"?

File: docs/source/services/key_value_stores/index.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+==================
+
+Overview
+========
+
+This matrix details the current state of available Key Value Services supported
+by the AtlasDB development staff.
+
++--------------------------------------+--------------------------------------+-------------------------------------------------------------+------------------------+------------------------------------------+----------------------------------------------+
+| Key Value Service                    | State                                | Service Versions Supported                                  | Recommended Versions   | Known Issues                             | Details                                      |
```
Comment (gsheasby) [inline docs/source/services/key_value_stores/index.rst]: I'd recommend using the "other" way of writing tables in rst, it looks way nicer.

File: docs/source/services/key_value_stores/index.rst [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
++--------------------------------------+--------------------------------------+-------------------------------------------------------------+------------------------+------------------------------------------+----------------------------------------------+
+| CQL KVS                              | Unsupported                          | ?                                                           | ?                      |                                          | :ref:`cql-kvs`                               |
++--------------------------------------+--------------------------------------+-------------------------------------------------------------+------------------------+------------------------------------------+----------------------------------------------+
+| JDBC KVS                             | Unsupported                          | ?                                                           | ?                      | #371                                     | :ref:`jdbc-kvs`                              |
```
Comment (gsheasby) [inline docs/source/services/key_value_stores/index.rst]: Link from #371 to github?

File: docs/source/services/key_value_stores/index.rst [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+Overview
+========
+
+This matrix details the current state of available Key Value Services supported
+by the AtlasDB development staff.
+
++--------------------------------------+--------------------------------------+-------------------------------------------------------------+------------------------+------------------------------------------+----------------------------------------------+
+| Key Value Service                    | State                                | Service Versions Supported                                  | Recommended Versions   | Known Issues                             | Details                                      |
++======================================+======================================+=============================================================+========================+==========================================+==============================================+
+| Cassandra KVS                        | Supported                            | Everything: Cassandra 2.0+ Data tables only: Cassandra 1.2+ | 2.2.8                  | none                                     | :ref:`cassandra-kvs`                         |
```
Comment (gsheasby) [inline docs/source/services/key_value_stores/index.rst]: Needs some kind of clear separation between `2.0+` and `Data`.

**Comment by rhero** [comment]
I cleaned up a lot of things.

- don't really like the issues column or the details column, removed them
- linked to config from the table
- changed naming of config pages to be consistent with the actual KVS name
- moved configs of KVSs to their own folder
- added words to the details part of the services page
- reformatted table to be the same as our release notes and most other tables in the docs

**Comment by jboreiko** [comment]
@gsheasby want to make your changes here and merge?

**Comment by gsheasby** [comment]
@jboreiko added to my todo list.

**Pull request merged**